libr/core/cmd_print.c

```diff
@@ -4486,6 +4486,111 @@ static void r_core_disasm_table(RCore * core, int l, const char *input) {
 	r_table_free (t);
 }
 
+static void cmd_pxr(RCore *core, int len, int mode, int wordsize, const char *arg) {
+	PJ *pj = NULL;
+	RTable *t = NULL;
+	if (mode == ',') {
+		t = r_table_new ();
+		RTableColumnType *n = r_table_type ("number");
+		RTableColumnType *s = r_table_type ("string");
+		r_table_add_column (t, n, "addr", 0);
+		r_table_add_column (t, n, "value", 0);
+		r_table_add_column (t, s, "refs", 0);
+	}
+	if (mode == 'j') {
+		pj = pj_new ();
+		if (!pj) {
+			return;
+		}
+	}
+	if (mode == 'j' || mode == ',' || mode == '*' || mode == 'q') {
+		size_t i;
+		const int base = core->anal->bits;
+		if (pj) {
+			pj_a (pj);
+		}
+		const ut8 *buf = core->block;
+		int withref = 0;
+		for (i = 0; i + wordsize < core->blocksize; i += wordsize) {
```

Comment:
shouldn't this be `i < core->blocksize` ? `i += wordsize` is always executed at each loop before the loop condition.

Reply:
i'm trying to avoid parsing incomplete words. so the check is correct imho

Reply:
oh i see the issue. yes it is correct the check.